### ISSUE TITLE
LeechBlockNG with E10S causes high CPU usage and page opening delay

### ISSUE THREAD

**Issue by ke352802081770314** [proginosko/LeechBlockNG#149]
Many thanks for maintaining LeechBlock.
(It was hinted at in issue #42 but since that mainly has another focus, I thought it is better to open a separate issue for this.)
I use Firefox on Ubuntu, currently 77.0.1. Only 2 extensions: LeechBlock NG and NoScript.
Until recently, I avoided the E10S (Firefox Electrolysis = multi-process architecture), first by using an about:config flag, after that stopped working, by setting the environment variable MOZ_FORCE_DISABLE_E10S=true and LeechBlockNG was working flawlessly and without noticeable performance impact for me in the single-process Firefox. Unfortunately, multiple relevant sites like "dashboards" including world maps and increasingly often Google Maps started causing an immediate browser crash when opening such a page, and I suspected (it seems rightly) that the single-process version choked on them because it is no longer supported by Firefox and not tested (also, about:performance simply stayed an empty page, making identifying CPU-hogs harder).
Since I've switched to multi-process, I notice in Task Manager that the process "WebExtensions" sticks at 12% (I have an i7 quadcore, so Linux sees 8 "CPUs" and 12% means one CPU/core/thread is staying at 100% processing) for several seconds after I open a page in a new tab, or enter an URL in an existing tab, while Firefox displays the moving grey dots in the favicon place of the tab, indicating that the page has not yet started loading content (they become blue dots while the page is loaded/rendered), and a "Web Content" process is at 1%. 
Also, when switching from another tab to an already open LeeckBlock Options tab, the tab is unresponsive for a number of seconds, e.g. I click on a weekday checkbox in the block set that is displayed and only after 5 to 10 seconds it is toggled visibly, until that happens, also the scrollbar does nothing.
So it seems that LeechBlock performance suffers from the interprocess communication between the "WebExtensions" process and some other Firefox process (I am not familiar with the inner workings of Firefox processes I just observe what I can see).
I have 12GB RAM and memory is not an issue, free says:
```
$ free
              total        used        free      shared  buff/cache   available
Mem:       12171024     5507408      929532     1007272     5734084     5335500
Swap:      12488700        2120    12486580
```
My Firefox has typically 9 windows with a total of about 1700 tabs most of which haven't yet been visited since the last Firefox start, and LeeckBlock is set to "Process only active tabs (improves performance when multiple tabs open)", and two Block Sets in use, both with "Count time spent on these sites only when browser tab is active", one with 5 lines of domain terms, the other with 1826 terms, I'm not using the ~ feature. 
Since it was unnoticeably fast in single-process mode, I think the real processing takes minimal time, so the delay must be due to some sort of overhead or temporary deadlock in process communication. I hope this helps you track down the issue.
